homeassistant/components/weatherflow_cloud/sensor.py

```diff
@@ -0,0 +1,239 @@
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_count_last_3hr",
+        translation_key="lightning_strike_count_last_3hr",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.observation.obs[0].lightning_strike_count_last_3hr,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_last_distance",
+        translation_key="lightning_strike_last_distance",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        value_fn=lambda data: data.observation.obs[0].lightning_strike_last_distance,
+    ),
+    WeatherFlowCloudSensorEntityDescription(
+        key="lightning_strike_last_epoch",
+        translation_key="lightning_strike_last_epoch",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: datetime.fromtimestamp(
+            data.observation.obs[0].lightning_strike_last_epoch, tz=UTC
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up WeatherFlow sensors based on a config entry."""
+
+    coordinator: WeatherFlowCloudDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    stations = coordinator.data.keys()
+
+    for sensor_description in WF_SENSORS:
+        for station_id in stations:
+            async_add_entities(
+                [
+                    WeatherFlowCloudSensor(
+                        coordinator=coordinator,
+                        description=sensor_description,
+                        station_id=station_id,
+                    )
+                ],
+                update_before_add=True,
+            )
+
+
+class WeatherFlowCloudSensor(WeatherFlowCloudEntity, SensorEntity):
+    """Implementation of a WeatherFlow sensor."""
+
+    entity_description: WeatherFlowCloudSensorEntityDescription
+
+    @property
+    def native_value(self) -> StateType | date | datetime | Decimal | None:
+        """Return the state of the sensor."""
+        return self.entity_description.value_fn(self.coordinator.data[self.station_id])
+
+    @property
+    def extra_state_attributes(self) -> dict[str, str] | None:
+        """Return the state attributes."""
+        if self.entity_description.extra_state_attributes_fn:
+            return self.entity_description.extra_state_attributes_fn(
+                self.coordinator.data[self.station_id]
+            )
+        return None
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon to use in the frontend, if any."""
+        if self.entity_description.icon_fn:
+            return self.entity_description.icon_fn(
+                self.coordinator.data[self.station_id]
+            )
+        return super().icon
```

Comment:
Haven't seen `icon_fn` declared. Let's boot it

Reply:
We have decided its declared - closing